### ISSUE TITLE
Failed to activate the existdb package

### ISSUE THREAD

**Issue by mark-donoghue** [eXist-db/atom-existdb#26]
[Enter steps to reproduce:]

1. install existdb package from package manager
2. restart Atom

**Atom**: 1.21.1 x64
**Electron**: 1.6.15
**OS**: Mac OS X 10.12.6
**Thrown From**: [existdb](https://github.com/wolfgangmm/atom-existdb) package 0.6.0

### Stack Trace

Failed to activate the existdb package

```
At Attempt to change mapping for "tei" extension from "application/tei xml" to "application/xml". Pass `force=true` to allow this, otherwise remove "tei" from the list of extensions for "application/xml".

Error: Attempt to change mapping for "tei" extension from "application/tei xml" to "application/xml". Pass `force=true` to allow this, otherwise remove "tei" from the list of extensions for "application/xml".
    at Mime.define (/packages/existdb/node_modules/mime/Mime.js:32:17)
    at /packages/existdb/lib/existdb-tree-view.coffee:22:18)
    at Object.activate (/packages/existdb/lib/existdb.coffee:38:25)
    at Package.module.exports.Package.activateNow (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package.js:253:25)
    at /Users/username/Applications/Atom.app/Contents/Resources/app/src/package.js:225:38
    at Package.module.exports.Package.measure (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package.js:99:21)
    at /Users/username/Applications/Atom.app/Contents/Resources/app/src/package.js:218:32
    at Package.module.exports.Package.activate (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package.js:215:40)
    at PackageManager.activatePackage (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:690:42)
    at config.transactAsync (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:665:36)
    at Config.module.exports.Config.transactAsync (/Users/username/Applications/Atom.app/Contents/Resources/app/src/config.js:367:24)
    at PackageManager.activatePackages (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:663:23)
    at PackageManager.activate (/Users/username/Applications/Atom.app/Contents/Resources/app/src/package-manager.js:646:50)
    at /Users/username/Applications/Atom.app/Contents/Resources/app/src/atom-environment.js:854:36
```

### Commands

```
```

### Non-Core Packages

```
atom-ide-ui 0.5.1 
existdb 0.6.0 
minimap 4.29.7 
```

**Comment by joewiz** [comment]
@mark-donoghue The public-repo's version of atom-editor-support, v0.2, is out of date. The current version, which ships with the atom-existdb package and which atom should prompt you to detect it connects to an eXist-db instance that doesn't have it installed, is v0.2.7. I'd suggest uninstalling v0.2 via the Dashboard, and then in Atom selecting Packages > existdb > Reconnect. This should cause Atom to detect that the atom-editor-support app isn't installed and offer to install the bundled v0.2.7 version.

**Comment by ttasovac** [comment]
hey @joewiz, I was so happy to see your instructions, but I'm afraid they didn't do the the trick for me. Installing the exist-db package alongside an exist instance without the atom-editor-support package doesn't actually install the atom-editor-support at all. 

I installed the v.0.2.7 manually, but even with it there, I'm still getting the same error...

**Comment by ttasovac** [comment]
Wolfgang, do you have any idea how to fix this?

**Comment by dizzzz** [comment]
same with atom v1.23.3 and plugin v0.6.0

**Comment by stefanprobst** [comment]
This particular error was fixed with 5242a7c, which unfortunately is not included in the latest release. For now you can follow the instructions on [building from source](https://github.com/wolfgangmm/atom-existdb#building-from-source)

**Comment by welblaud** [comment]
I have just uploaded the last version of existdb Atom support package from build (0.2.7) but it throws still the same. In Atom I have the version 0.6.0 of the atom-existdb package. I need to connect to an external server via https. Could it be the problem? (I have set the URL in atom-existdb config.)

**Comment by joewiz** [comment]
I can confirm that the original issue as reported remains valid with the current version of atom, 1.24.0. When trying to install the existdb package via Atom > Preferences > Install > existdb > Install (the current release listed is 0.6.0), installation fails with:

> Failed to activate the existdb package. Attempt to change mapping for "tei" extension from "application/tei+xml" to "application/xml". Pass `force=true` to allow this, otherwise remove "tei" from the list of extensions for "application/xml". The error was thrown from the existdb package. This issue has already been reported.

It appears that several of Wolfgang's [commits since the 0.6.0 release](https://github.com/wolfgangmm/atom-existdb/compare/v0.6.0...master) address compatibility problems with Atom 1.21.0. 

In the short term, installing the package [built from source](https://github.com/wolfgangmm/atom-existdb#building-from-source) and restarting Atom does prevent this error (i.e., the one that @welblaud reported above). The version of the companion application for eXist (atom-editor-support) that is included in the version of the atom-existdb atom package built from source is 0.2.7 and was [committed on June 15](https://github.com/wolfgangmm/atom-existdb/blob/master/resources/db/atom-editor-0.2.7.xar) (much newer than the version in the exist-db.org public-repo, 0.2.2). There have been some important fixes made in [commits to the atom-editor-support since June 15](https://github.com/wolfgangmm/atom-editor-support/compare/e58b519d737a9e89991a6f6fdd416c1690bd87bd...master?w=1).

So to sum up, I think a new release of both the atom-exist atom package and the atom-editor-support eXist app are in order. 

Wolfgang, I'd be happy to assist in the preparation of these updates if it would help. The best way to give me the necessary permissions for committing and publishing may be for you to migrate the repositories to the eXist-db organization as discussed in recent Community Calls.

**Comment by welblaud** [comment]
For me it does not work. I use Atom 1.25 (beta), the package atom-existdb is 0.6.0, in eXist (3.6.0) I have the builded packate 0.2.7. Should I restart the server? Or could it be associated with the beta version of Atom? (I have to use it because I need the portable version.)

**Comment by stefanprobst** [comment]
You need to also build a new version of atom-existdb, since 0.6.0 will not work. Just follow the [instructions](https://github.com/wolfgangmm/atom-existdb#building-from-source). The problem is that one of the dependencies, the [mime](https://www.npmjs.com/package/mime) package, had API-breaking changes, that were fixed with 5242a7ce36b12b23542404709ffd72bb9d267bf7.

**Comment by welblaud** [comment]
Thank you! It somehow helped and Atom started to solve dependencies. The last thing (strange!) was it asked whether I wish to install server side support app. The strange part was after the question it tried to show the app version but instead of the version numbers it logged HTML of the main page of my app (Home).

When I agreed to install the support despite the strange log, it threw:

### Stack Trace

Uncaught TypeError: Cannot read property 'onSelect' of undefined

```
At D:\C_temp\Atom\exist-db-package\atom-existdb\lib\existdb-tree-view.coffee:173

TypeError: Cannot read property 'onSelect' of undefined
    at Request._callback (D:/C_temp/Atom/exist-db-package/atom-existdb/lib/existdb-tree-view.coffee:173:39)
    at Request.self.callback (/packages/existdb/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:194:7)
    at /packages/existdb/node_modules/request/request.js:1163:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:191:7)
    at /packages/existdb/node_modules/request/request.js:1085:12)
    at Object.onceWrapper (events.js:293:19)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:188:7)
    at endReadableNT (_stream_readable.js:975:12)
    at _combinedTickCallback (internal/process/next_tick.js:80:11)
    at process._tickCallback (internal/process/next_tick.js:104:9)
```

### Commands

```
     -1:21.9.0 core:delete (input.hidden-input)
  3x -1:21.5.0 core:move-right (input.hidden-input)
     -1:20.8.0 editor:select-to-end-of-line (input.hidden-input)
  2x -1:20.6.0 core:select-left (input.hidden-input)
     -1:19.8.0 intentions:highlight (input.hidden-input)
     -1:19.6.0 core:paste (input.hidden-input)
 29x -1:17.9.0 core:move-left (input.hidden-input)
     -1:16 core:move-right (input.hidden-input)
  7x -1:15.7.0 core:backspace (input.hidden-input)
     -1:14.2.0 intentions:highlight (input.hidden-input)
     -1:13.8.0 core:save (input.hidden-input)
     -1:11.7.0 editor:select-to-end-of-line (input.hidden-input)
  2x -1:11.4.0 core:select-left (input.hidden-input)
  7x -1:09.0 core:backspace (input.hidden-input)
     -0:59.3.0 intentions:highlight (input.hidden-input)
     -0:58.8.0 core:save (input.hidden-input)
```

### Non-Core Packages

```
atom-ide-ui 0.9.2 
atom-material-syntax 1.0.8 
atom-material-syntax-dark 1.0.0 
atom-material-ui 2.1.3 
atom-runner 2.7.1 
atom-ternjs 0.18.3 
busy-signal 1.4.3 
caprice-dark-syntax 2.0.0 
deep-purple-syntax 0.2.18 
duotone-dark-sky-syntax 2.1.0 
existdb 0.7.0 
hyperclick 0.0.0 
ide-java 0.8.1 
ide-typescript 0.7.5 
idle-theme 0.1.0 
intentions 1.1.5 
joehannes-steampunk-ui 0.2.11 
linter 2.2.0 
linter-ui-default 1.6.10 
monokai 0.24.0 
steam-pirate-ui 0.6.1 
```

**Comment by wolfgangmm** [comment]
I moved this repo and atom-editor-support to the eXist-db github and published a new release. For some reason apm assigned it version 0.8.0 instead of 0.7.0 (no idea why), but I hope this fixes the issue.

**Comment by joewiz** [comment]
Closing for now, as the original issue is no longer evident with version 0.8.0 of the atom-existdb package (install via Atom's preferences > Packages). Please open a new issue for any problems with the new version.
